code/scripting/api/libs/base.cpp

```diff
@@ -229,7 +229,17 @@ ADE_FUNC(getFrametimeOverall, l_Base, NULL, "The overall frame time in seconds s
 	return ade_set_args(L, "x", game_get_overall_frametime());
 }
 
-ADE_FUNC(getFrametime, l_Base, "[boolean adjustForTimeCompression]", "Gets how long this frame is calculated to take. Use it to for animations, physics, etc to make incremental changes.", "number", "Frame time (seconds)")
+ADE_FUNC(getTimeCompressedFrametime, l_Base, NULL, "Gets how long this frame is calculated to take. Use it to for animations, physics, etc to make incremental changes. Increased or decreased based on current time compression", "number", "Frame time (seconds)")
+{
+	return ade_set_args(L, "f", flFrametime);
+}
+
+ADE_FUNC(getRealFrametime, l_Base, NULL, "Gets how long this frame is calculated to take in real time. Not affected by time compression.", "number", "Frame time (seconds)")
+{
+	return ade_set_args(L, "f", flRealframetime);
+}
+
+ADE_FUNC(getFrametime, l_Base, "[boolean adjustForTimeCompression]", "Gets how long this frame is calculated to take. Use it to for animations, physics, etc to make incremental changes. DEPRECATED, due a mistake in the documentaion.", "number", "Frame time (seconds)")
```

Comment:
You should add a warning that only shows once per game session if this one is used to inform users and modders that this script needs to be updated.

Reply:
Note that, it is entirely possible to be using this function correctly (the fault is really just with the documentation), and that until the MVPs is updated this will be warning everyone who uses or depends on it. Given this I really fear that a warning is over-aggressive.

Reply:
Then we need to poke the MVPs people aggressively to fix this ASAP instead.  Let's see what other people have to say about this PR.

Reply:
This should use `ADE_FUNC_DEPRECATED` which will handle the deprecation behavior automatically. The deprecation version should be the next release so `20.2`.

That will mean that there will be a message in the API doc specifying that the function is deprecated and the function will also not be available when the mod reports a supported engine version equal or higher than the deprecation version.

Reply:
Ah, I was wondering how to handle deprecation.